hazelcast/src/main/java/com/hazelcast/map/impl/BiFunctionExecutingEntryProcessor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.map.EntryProcessor;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+public class BiFunctionExecutingEntryProcessor<K, V> implements EntryProcessor<K, V, V> {
```

Comment:
It would be great if you could make this class implement `IdentifiedDataSerializable` since currently this class uses java serialisation which is not efficient and does not allow class evolution. It does get complicated here, though. Once you add the interface, you'll have to implement some additional methods, like:

```
@Override
    public int getFactoryId() {
        return MapDataSerializerHook.F_ID;
    }

    @Override
    public int getClassId() {
        return MapDataSerializerHook.BI_FUNCTION_EXECUTING_EP;
    }

    @Override
    public void writeData(ObjectDataOutput out) throws IOException {
        out.writeObject(biFunction);
    }

    @Override
    public void readData(ObjectDataInput in) throws IOException {
        biFunction = in.readObject();
    }
```

This will also require adding a new row in `MapDataSerializerHook` but also a new client message and you'll need to modify the client protocol. I suggest you look at the `EntryRemovingProcessor` class and how it's instantiated and used from other parts of the code.

Reply:
Hi @mmedenjak , I am done with all changes. Just that I am not sure if/why would I need to add a new client message and modify the client protocol. I am using existing operations in the implementation for `ClientMapProxy.java`, which already have an existing implementation ... am I missing something?

Reply:
Also wanted to check, do I need to provide this implementation for clients in other languages too?

Reply:
Looks good to me. You don't need to provide an implementation for other languages.